main_a_youtube_to_mp3.py

```diff
@@ -1,29 +1,46 @@
 import ssl
 import argparse
 from pytube import YouTube
+from pytube import Search
 from moviepy.editor import *
 
 import config
 
 
-def run(url: str, out_dir: str, out_mp3_filename: str):
-    ssl._create_default_https_context = ssl._create_stdlib_context
-
-    yt = YouTube(url)
+def run(input: list, song_name: bool, out_dir: str, out_mp3_filename: str):
+    ssl._create_default_https_context = ssl._create_stdlib_context 
+
+    if song_name == False:
+        yt = YouTube(input[0])
+    else:
+        print('Searching...')
+        query = " ".join(input)
+        try:
+            s = Search(query)
+        except:
+            # to be more specific with exceptions (https://github.com/pytube/pytube/blob/master/pytube/exceptions.py)
+            print('Search failed')
+            return
+        # choose the first video
+        yt = YouTube(s.results[0].watch_url)
```

Comment:
try restructure it into this pattern. it's generally better to shorten if/else block when applicable

```suggestion
    if not song_name:
        youtube_url = input[0]
    else:
        print('Searching...')
        query = " ".join(input)
        try:
            s = Search(query)
        except:
            # to be more specific with exceptions (https://github.com/pytube/pytube/blob/master/pytube/exceptions.py)
            print('Search failed')
            return
        # choose the first video
        youtube_url = s.results[0].watch_url
        
    yt = YouTube(youtube_url)
```